packages/salesforce-adapter/src/filters/workflow.ts

```diff
@@ -149,20 +149,17 @@ const createDummyWorkflowInstance = async (
 const createWorkflowChange = async (
   changes: ReadonlyArray<Change<InstanceElement>>,
 ): Promise<Change<InstanceElement>> => {
-  const workflowChange = await awu(changes).find(isInstanceOfTypeChange(WORKFLOW_METADATA_TYPE))
-
-  const parent = workflowChange === undefined
-    ? await createDummyWorkflowInstance(changes)
-    : getChangeElement(workflowChange)
+  const parent = await createDummyWorkflowInstance(changes)
   const after = await createPartialWorkflowInstance(parent, changes, 'after')
-  if (workflowChange !== undefined && isAdditionChange(workflowChange)) {
-    return { action: 'add', data: { after } }
-  }
   // we assume the only possible changes are in nested instances and we omit the nested instance
   // fields from the parent when we create the partial instance. so we can use the same parent as
   // the after instance here
   const before = await createPartialWorkflowInstance(parent, changes, 'before')
-
+  /*
+   * we won't be able to know if the change was add or modified
+   * it does not matter in this use case since changes
+   * will go to salesforce upsert or add API anyway.
+   */
```

Comment:
`upsert` is already `update or insert`, doesn't make sense to say `upsert or add`

```suggestion
  // we cannot know if the workflow instance change is add or modify
  // but it does not matter in this use case because changes
  // will be deployed with the upsert API anyway
```

also, now that we always use the dummy instance as a parent, the comment in lines 154-156 is not needed anymore and can be removed